utils/s2n_random.c

```diff
@@ -327,30 +446,14 @@ S2N_RESULT s2n_set_private_drbg_for_test(struct s2n_drbg drbg)
     return S2N_RESULT_OK;
 }
 
-
-bool s2n_cpu_supports_rdrand()
-{
-#if ((defined(__x86_64__) || defined(__i386__)) && (defined(__clang__) || S2N_GCC_VERSION_AT_LEAST(4,3,0)))
-    uint32_t eax, ebx, ecx, edx;
-    if (!__get_cpuid(1, &eax, &ebx, &ecx, &edx)) {
-        return false;
-    }
-
-    if (ecx & RDRAND_ECX_FLAG) {
-        return true;
-    }
-#endif
-    return false;
-}
-
 /*
  * volatile is important to prevent the compiler from
  * re-ordering or optimizing the use of RDRAND.
  */
-S2N_RESULT s2n_get_rdrand_data(struct s2n_blob *out)
+static int s2n_rand_rdrand_impl(void *data, uint32_t size)
```

Comment:
The all zero/all one check needs to lifted out of the platform `ifdef` and always check those values. (The CPU bug it is defending against can impact 64 bit processors as well).

Reply:
Unless I'm mistaken, the check is there (outside the `ifdef`) for 64 bit processors:

https://github.com/awslabs/s2n/blob/d64e63bd5632168fcddd25f2cf64df62fa3deffd/utils/s2n_random.c#L414-L430

Reply:
Ah. I see it. I was confused by the 32 bit check (which will work for 64 as well, just with a slight risk of false positives) not being elsewhere. (I'm still getting used to the s2n code-base and coding style.)